src/PIL/PpmImagePlugin.py

```diff
@@ -110,6 +113,14 @@ def _open(self):
         if magic_number in (b"P1", b"P2", b"P3"):
             decoder_name = "ppm_plain"
         for ix in range(3):
+            if mode == "F" and ix == 2:
+                scale = float(self._read_token())
+                if not math.isfinite(scale) or scale == 0.0:
```

Comment:
I find it a bit more readable to have `not` last:
```suggestion
                if scale == 0.0 or not math.isfinite(scale):
```